components/Intro.tsx

```diff
@@ -1,30 +1,84 @@
 import config from "config";
+import { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { setAdmin } from "redux/slices/admin";
+import { useAppSelector } from "redux/typesHooks";
 
-const Intro = () => (
-  <section className="intro-section">
-    <div className="container">
-      <div className="intro-content">
-        <div className="intro-avatar">
-          <div className="image">
-            <img
-              src="/images/author-avatar.jpg"
-              alt="author-avatar"
-              width={90}
-              height={90}
-            />
+const Intro = () => {
+  const dispatch = useDispatch();
+  const admin = useAppSelector((state) => state.admin);
+
+  const [isFirstRender, setIsFirstRender] = useState(true);
+
+  const handleClick = () => {
+    if (admin === true) {
+      dispatch(setAdmin(false));
+    } else {
+      dispatch(setAdmin(true));
+    }
+  };
+
+  useEffect(() => {
+    const imageWrapper = document.querySelector("div.image-avatar");
+    const editIcon = document.querySelector("img.editor-icon");
+
+    if (admin) {
+      imageWrapper?.setAttribute("admin", "");
+      editIcon?.setAttribute("admin", "");
+      setIsFirstRender(false);
+    }
+    if (!admin && isFirstRender === false) {
+      imageWrapper?.setAttribute("closing", "");
+      editIcon?.setAttribute("closing", "");
+      imageWrapper?.addEventListener(
+        "animationend",
+        () => {
+          editIcon?.removeAttribute("closing");
+          imageWrapper?.removeAttribute("closing");
+        },
+        { once: true }
+      );
+      imageWrapper?.removeAttribute("admin");
+      editIcon?.removeAttribute("admin");
+    }
+  }, [admin, isFirstRender]);
+
+  return (
+    <section className="intro-section">
```

Comment:
https://github.com/1alexvash/AlexCodeBlog/pull/82 - refactor of MUI component. I think it will be easier for you to check this refactor if it will be like another PR.